components/Snackbar/src/MDCSnackbarManager.m

```diff
@@ -254,7 +254,7 @@ - (void)displaySnackbarViewForMessage:(MDCSnackbarMessage *)message {
                                                 message.voiceNotificationText);
               }
 
-              if ([self isSnackbarTransient:snackbarView]) {
+              if ([self isSnackbarTransient:snackbarView] && message.duration > 0) {
```

Comment:
I understand that this is a replication of an internal API, but trying to put myself in the shoes of someone unfamiliar with Material Snackbars, setting duration to a non-positive value seems like it would lead to undefined behavior. I think I'd prefer a separate property on `MDCSnackbarMessage` to indicate whether it should automatically dismiss. Potentially `automaticallyDismisses` or `dismissesAfterDuration`?

Reply:
yeah, i pondered on that for a bit. I didn't like the overwhelming amount of properties the Snackbar and other components of ours have, and I thought the client's solution was elegant, if well documented. I don't feel strongly so will move to a opt-in property.